UI/src/model/locationSearch.js

```diff
@@ -29,92 +29,95 @@
+          var y = _.get(speedlimitdata, 'latitude');
+          var x= _.get(speedlimitdata, 'longitude');
+          var title = input.text + " (nopeusrajoituksen ID)";
             returnObject.push({title: title, lon: x, lat: y, linkid:linkid, resultType:"SpeedLimit"});
         }
-        if (returnObject.length===0){
+
+        if (_.isEmpty(returnObject))
           return $.Deferred().reject('Haulla ei löytynyt tuloksia');
-        }
         return returnObject;
         });
     };
 
     /**
-     * Link id search, combined with road number search
+     * Link id search
      *
      * @param input
      * @returns {*}
      */
-    var roadNumberAndRoadLinkSearch= function(input) {
-      var roadLinkSearch = backend.getRoadLinkToPromise(input.text);
-      var roadNumberSearch = backend.getCoordinatesFromRoadAddress(input.text);
-      return $.when(roadLinkSearch, roadNumberSearch).then(function(linkdata,roadData) {
-        var returnObject = roadLocationAPIResultParser(roadData);
-        if (_.get(linkdata[0], 'success')) {
-          var x = _.get(linkdata[0], 'middlePoint.x');
-          var y = _.get(linkdata[0], 'middlePoint.y');
+    var roadLinkSearchById= function(input) {
+      return $.when(backend.getRoadLinkToPromise(input.text)).then(function(linkdata) {
+        var returnObject = [];
+        if (_.get(linkdata, 'success')) {
+          var x = _.get(linkdata, 'middlePoint.x');
+          var y = _.get(linkdata, 'middlePoint.y');
           var title = input.text + " (linkin ID)";
           returnObject.push({title: title, lon: x, lat: y, resultType: "Link-id"});
         }
-          if (returnObject.length === 0){
+
+        if (_.isEmpty(returnObject))
           return $.Deferred().reject('Haulla ei löytynyt tuloksia');
-          }
         return returnObject;
       });
     };
 
+
+    var extractSearchResultInfo = function (massTransitStop) {
+      return { lon: _.get(massTransitStop, 'lon'),
+        lat: _.get(massTransitStop, 'lat'),
+        nationalId: _.get(massTransitStop, 'nationalId')
+      };
+    };
+
     /**
-     * Mass transit stop national id search, combined with road number search
+     * Mass transit stop national id search
      *
      * @param input
      * @returns {*}
      */
-    var roadNumberAndNationalIdSearch = function(input) {
-      return $.when(backend.getMassTransitStopByNationalIdForSearch(input.text), backend.getCoordinatesFromRoadAddress(input.text)).then(function(result,roadData) {
-        var returnObject = roadLocationAPIResultParser(roadData);
-         if (_.get(result[0], 'success')) {
-          var lon = _.get(result[0], 'lon');
-          var lat = _.get(result[0], 'lat');
+    var massTransitStopSearchByNationalId = function(input) {
+      return $.when(backend.getMassTransitStopByNationalIdForSearch(input.text)).then(function(result) {
+        var returnObject = [];
+
+        if (_.get(result, 'success')) {
```

Comment:
Comparing to previous,  result is an array.
Shouldn't we validate if it is an array?

Reply:
Don´t know why but it was never a array but a object